staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -140,6 +141,16 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		clientConfig.Timeout = timeout
 	}
 
+	if configClusterInfo.ConnectionTimeout != "" {
+		connectionTimeout, err := ParseTimeout(configClusterInfo.ConnectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+		clientConfig.ConnectionTimeout = connectionTimeout
+	} else {
+		clientConfig.ConnectionTimeout = 7 * time.Second
```

Comment:
Just to flag: I'm not sure that this is the correct way to apply a default value – as far as I can see there is no mechanism for defaulting any of the other values at present.